src/esp/io/JsonEspTypes.cpp

```diff
@@ -49,6 +49,16 @@ JsonGenericValue toJsonValue(const gfx::replay::Keyframe& keyframe,
     io::addMember(obj, "userTransforms", userTransformsArray, allocator);
   }
 
+  if (keyframe.lights && !keyframe.lights->empty()) {
```

Comment:
How does this logic differentiate an empty vector (a light setup with no lights) from a `nullopt` (no light setup)?

Reply:
The `Optional` `bool()` operator [is overridden to check whether the container has a value](https://doc.magnum.graphics/corrade/classCorrade_1_1Containers_1_1Optional.html#a389cd0c17eac32b9194f922fccc91934), much like [the implementation of `std::optional`](https://en.cppreference.com/w/cpp/utility/optional/operator_bool).